SCons/Environment.py

```diff
@@ -1051,7 +1051,8 @@ def get_CacheDir(self):
         cachedir_class = self.validate_CacheDir_class()
         try:
             if (path == self._last_CacheDir_path
-                    and cachedir_class is type(self._last_CacheDir)):
+                    and issubclass(type(self._last_CacheDir), cachedir_class)
```

Comment:
Do you need the type()'s here? shouldn't it work without?
If not, stick a comment here explaining what you're doing?
If I'm reading it right, you're verifying that they are the exact same class?

Reply:
the type needs to be there, issubclass compares class, which are types.

The two issubclass with args switched is making sure the two classes are the same class. It does the same thing as `type(self._last_CacheDir) is cachedir_class` but flake8 doesn't mind this version of it.

Honestly I think this is just making it less readable. I think the flake8 rule is flawed in its assumptions. Why would you use `type` if you want to allow checks against base classes as isinstance does, you would just use isinstance, and if you are using type then that means you **do** want an exact match.

Reply:
Fair enough. I guess return to original, but put a comment there explaining what and why it's doing that?
I'll figure out a way to disable that warning for that line.

Reply:
okay apparently slider is happy now with using type, did you disable that rule I suppose?

Reply:
hmm. nope. Did you have `== type()` vs now `is type(`?

Reply:
no I had the `is` before as well but I did switch the order of the args from https://github.com/SCons/scons/pull/3894/commits/ce9ea2580fa5e4bb143d5ada308c42e40512be19